KubeArmor panics when Kubernetes pods restart

KubeArmor panics when trying to detect new pods after restarting Kubernetes pods. 

### Steps to reproduce

Get [multiubuntu](https://github.com/accuknox/KubeArmor/blob/master/examples/multiubuntu.md) pods up.

Run KubeArmor.

**vagrant@kubearmor-dev:~/KubeArmor/KubeArmor$** `make run`

It detects the whole thing.

```bash
2021-06-24 13:16:18.278133	INFO	Started to serve gRPC-based log feeds
2021-06-24 13:16:18.281614	INFO	Initializing an eBPF program
2021-06-24 13:16:19.872862	INFO	Initialized the eBPF program
2021-06-24 13:16:20.091298	INFO	Started to monitor system events
2021-06-24 13:16:20.164983	INFO	Removed an inactive AppArmor profile (apparmor-demo-ubuntu-1)
2021-06-24 13:16:20.183571	INFO	Removed an inactive AppArmor profile (apparmor-demo-ubuntu-2)
2021-06-24 13:16:20.202706	INFO	Removed an inactive AppArmor profile (apparmor-demo-ubuntu-3)
2021-06-24 13:16:20.221855	INFO	Removed an inactive AppArmor profile (apparmor-demo-ubuntu-4)
2021-06-24 13:16:20.240342	INFO	Removed an inactive AppArmor profile (apparmor-demo-ubuntu-5)
2021-06-24 13:16:20.240493	INFO	Initialized AppArmor Enforcer
2021-06-24 13:16:20.240508	INFO	Started to protect containers
2021-06-24 13:16:21.273194	INFO	Initialized the Kubernetes client
2021-06-24 13:16:21.273244	INFO	Started to monitor Pod events
2021-06-24 13:16:21.273259	INFO	Started to monitor security policies
2021-06-24 13:16:21.287054	INFO	Detected a Pod (added/kube-system/kube-flannel-ds-snh55)
2021-06-24 13:16:21.287316	INFO	Detected a Pod (added/kube-system/kube-proxy-8zm7n)
2021-06-24 13:16:21.289736	INFO	Container Runtime: docker://18.3.1
2021-06-24 13:16:21.289853	INFO	Started to monitor Docker events
2021-06-24 13:16:21.293524	INFO	Detected a Pod (added/multiubuntu/ubuntu-4-deployment-fbd564bdc-m8xlj)
2021-06-24 13:16:21.330298	INFO	Registered an AppArmor profile (apparmor-demo-ubuntu-4)
2021-06-24 13:16:21.330661	INFO	Detected a Pod (added/kube-system/etcd-kubearmor-dev)
2021-06-24 13:16:21.331051	INFO	Detected a Pod (added/kube-system/kube-controller-manager-kubearmor-dev)
2021-06-24 13:16:21.332897	INFO	Detected a Pod (added/multiubuntu/ubuntu-2-deployment-84bc87bcdb-h6rf8)
2021-06-24 13:16:21.352405	INFO	Registered an AppArmor profile (apparmor-demo-ubuntu-2)
2021-06-24 13:16:21.354408	INFO	Detected a Pod (added/multiubuntu/ubuntu-1-deployment-7c87796756-p58b5)
2021-06-24 13:16:21.374888	INFO	Registered an AppArmor profile (apparmor-demo-ubuntu-1)
2021-06-24 13:16:21.375415	INFO	Detected a Pod (added/kube-system/coredns-558bd4d5db-klq8c)
2021-06-24 13:16:21.375724	INFO	Detected a Pod (added/kube-system/kube-scheduler-kubearmor-dev)
2021-06-24 13:16:21.377704	INFO	Detected a Pod (added/multiubuntu/ubuntu-3-deployment-88fcf59b4-rf5tz)
2021-06-24 13:16:21.398504	INFO	Registered an AppArmor profile (apparmor-demo-ubuntu-3)
2021-06-24 13:16:21.400345	INFO	Detected a Pod (added/multiubuntu/ubuntu-5-deployment-bb7cff785-4hfgx)
2021-06-24 13:16:21.419591	INFO	Registered an AppArmor profile (apparmor-demo-ubuntu-5)
2021-06-24 13:16:21.419938	INFO	Detected a Pod (added/kube-system/coredns-558bd4d5db-cxjgz)
2021-06-24 13:16:21.420196	INFO	Detected a Pod (added/kube-system/kube-apiserver-kubearmor-dev)
2021-06-24 13:16:22.290555	INFO	Initialized KubeArmor
```

Now restart the multiubuntu pods.

**vagrant@kubearmor-dev:~/KubeArmor/examples/multiubuntu$** `kubectl -n multiubuntu rollout restart deploy`

```bash
2021-06-24 13:16:52.794668	INFO	Detected a Pod (added/multiubuntu/ubuntu-1-deployment-745b588989-zkpjb)
2021-06-24 13:16:52.797282	INFO	Increased the refCount (1 -> 2) of an AppArmor profile (apparmor-demo-ubuntu-1)
2021-06-24 13:16:52.819802	INFO	Detected a Pod (added/multiubuntu/ubuntu-2-deployment-7bcd4fd95-8hkhc)
2021-06-24 13:16:52.822182	INFO	Increased the refCount (1 -> 2) of an AppArmor profile (apparmor-demo-ubuntu-2)
2021-06-24 13:16:52.847756	INFO	Detected a Pod (added/multiubuntu/ubuntu-3-deployment-7d7fc59499-8l99k)
2021-06-24 13:16:52.850395	INFO	Increased the refCount (1 -> 2) of an AppArmor profile (apparmor-demo-ubuntu-3)
2021-06-24 13:16:52.863842	INFO	Detected a Pod (added/multiubuntu/ubuntu-4-deployment-8686d7ddbb-96t6q)
2021-06-24 13:16:52.866231	INFO	Increased the refCount (1 -> 2) of an AppArmor profile (apparmor-demo-ubuntu-4)
2021-06-24 13:16:52.871524	INFO	Detected a Pod (added/multiubuntu/ubuntu-5-deployment-55cb75c4d4-2sphf)
2021-06-24 13:16:52.873962	INFO	Increased the refCount (1 -> 2) of an AppArmor profile (apparmor-demo-ubuntu-5)
2021-06-24 13:16:53.696801	INFO	Detected a container (added/997bccd7c6f2)
2021-06-24 13:16:53.938046	INFO	Detected a container (added/aa95f4589b35)
2021-06-24 13:16:54.119448	INFO	Detected a container (added/b131ec4eacca)
2021-06-24 13:16:54.340243	INFO	Detected a container (added/e12ed145711d)
2021-06-24 13:16:54.561384	INFO	Detected a container (added/5e908e6f6315)
2021-06-24 13:16:55.474393	INFO	Detected a container (added/7e7c8dd91e27)
2021-06-24 13:16:56.043825	INFO	Detected a Pod (modified/multiubuntu/ubuntu-3-deployment-7d7fc59499-8l99k)
2021-06-24 13:16:56.751614	INFO	Detected a container (added/593d20913d69)
2021-06-24 13:16:57.059048	INFO	Detected a Pod (modified/multiubuntu/ubuntu-4-deployment-8686d7ddbb-96t6q)
2021-06-24 13:16:57.963738	INFO	Detected a container (added/aef2a275c3dc)
2021-06-24 13:16:58.078789	INFO	Detected a Pod (modified/multiubuntu/ubuntu-1-deployment-745b588989-zkpjb)
2021-06-24 13:16:59.275444	INFO	Detected a container (added/04bad8bf7f48)
2021-06-24 13:17:00.112302	INFO	Detected a Pod (modified/multiubuntu/ubuntu-2-deployment-7bcd4fd95-8hkhc)
2021-06-24 13:17:00.506709	INFO	Detected a container (added/31552a1f331d)
2021-06-24 13:17:01.168465	INFO	Detected a Pod (modified/multiubuntu/ubuntu-5-deployment-55cb75c4d4-2sphf)
panic: runtime error: slice bounds out of range [:12] with length 0

goroutine 58 [running]:
github.com/accuknox/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).UpdateDockerContainer(0xc0005665a0, 0xc001852440, 0x40, 0xc0022122ec, 0x4)
	/home/vagrant/KubeArmor/KubeArmor/core/dockerHandler.go:186 +0x658
github.com/accuknox/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).MonitorDockerEvents(0xc0005665a0)
	/home/vagrant/KubeArmor/KubeArmor/core/dockerHandler.go:215 +0x238
created by github.com/accuknox/KubeArmor/KubeArmor/core.KubeArmor
	/home/vagrant/KubeArmor/KubeArmor/core/kubeArmor.go:468 +0x82a
Makefile:12: recipe for target 'run' failed
make: *** [run] Error 2

```

Hi @geyslan . I am able to reproduce this issue without using make run. I deployed KubeArmor using the deployment yaml. 
```
2021-06-25 11:20:33.331801	INFO	Detected a Pod (modified/multiubuntu/ubuntu-1-deployment-78c6d6c5bc-qcrnp)
2021-06-25 11:20:33.376186	INFO	Updated 1 security rules to multiubuntu/ubuntu-1-deployment-78c6d6c5bc-qcrnp/apparmor-demo-ubuntu-1
2021-06-25 11:20:35.763924	INFO	Detected a container (added/97a30762e79c)
2021-06-25 11:20:36.384414	INFO	Detected a Pod (modified/multiubuntu/ubuntu-5-deployment-555455fbd6-sdgvw)
2021-06-25 11:20:38.656213	INFO	Detected a container (added/a810d59c7102)
2021-06-25 11:20:39.463801	INFO	Detected a Pod (modified/multiubuntu/ubuntu-4-deployment-766447b7b4-v5s4z)
panic: runtime error: slice bounds out of range [:12] with length 0

goroutine 68 [running]:
github.com/accuknox/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).UpdateDockerContainer(0xc00024b0e0, 0xc001a88740, 0x40, 0xc001756620, 0x4)
	/usr/src/KubeArmor/KubeArmor/core/dockerHandler.go:186 +0x658
github.com/accuknox/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).MonitorDockerEvents(0xc00024b0e0)
	/usr/src/KubeArmor/KubeArmor/core/dockerHandler.go:215 +0x278
created by github.com/accuknox/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:447 +0x52a
```
The key point of this issue is that multiubuntu must be deployed before KubeArmor. This issue is directly a consequence of #163.

--- a/KubeArmor/core/dockerHandler.go
+++ b/KubeArmor/core/dockerHandler.go
@@ -183,7 +183,7 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 		}
 		dm.ContainersLock.Unlock()
 
-		dm.LogFeeder.Printf("Detected a container (removed/%s)", container.ContainerID[:12])
+		dm.LogFeeder.Printf("Detected a container (removed/%s)", containerID[:12])
 	}
 }
 

